base/ca/src/com/netscape/cms/servlet/cert/scep/CRSEnrollment.java

```diff
@@ -408,12 +449,39 @@ public void service(HttpServletRequest httpReq,
 
     }
 
+    /**
+     *
+     * Extracts the ProfileId encoded in the URI.
+     * Expects URI in the form of "/scep/PROFILE_ID/pkiclient.exe".
+     *
+     * @param httpReq The HttpServletRequest.
+     *
+     */
+    private String extractProfileIdFromURL(HttpServletRequest httpReq) throws ServletException {
+        String pathInfo = httpReq.getPathInfo();
+
+        if (!pathInfo.matches("^/[^/]+/pkiclient\\.exe$")) {
+            throw new ServletException("dynamic ProfileId URL must be in the form of '" +
+                    httpReq.getContextPath() + httpReq.getServletPath() + "/PROFILE_ID/pkiclient.exe'");
+        }
+
+        return pathInfo.split("/")[1];
+    }
+
     private boolean isAlgorithmAllowed(String[] allowedAlgorithm, String algorithm) {
+        return isInAllowedList(allowedAlgorithm, algorithm);
+    }
+
+    private boolean isDynamicProfileIdAllowed(String[] allowedDynamicProfileId, String profileId) {
+        return isInAllowedList(allowedDynamicProfileId, profileId);
+    }
+
+    private boolean isInAllowedList(String[] allowedList, String searchedItem) {
         boolean allowed = false;
 
-        if (algorithm != null && algorithm.length() > 0) {
-            for (int i = 0; i < allowedAlgorithm.length; i++) {
-                if (algorithm.equalsIgnoreCase(allowedAlgorithm[i])) {
+        if (searchedItem != null && searchedItem.length() > 0) {
+            for (int i = 0; i < allowedList.length; i++) {
+                if (searchedItem.equalsIgnoreCase(allowedList[i])) {
```

Comment:
I think profile ID should be case sensitive, so we should use `equals()` instead of `equalsIgnoreCase()`.

Reply:
I'd agree on principal, but at least one other place in PKI use `equalsIgnoreCase`: `base/ca/src/com/netscape/ca/CRLIssuingPoint.java:696`

I do think we should unify and reach one version.

Reply:
also, this is shared with the isAlgorithmAllowed() call, which is not case sensitive.  I'll leaves it as is then.  Thanks!